### DISCUSSION TITLE
enable CONFIG_NO_HZ_FULL=y

### DISCUSSION THREAD

**Discussion by unclehack** [home-assistant/operating-system#3702] [Ideas]
Switching the kernel to full tickless at least for x86_64 would help reduce latency for various operations and power consumption.

**Comment by sairon** [comment]
Our kernel uses `CONFIG_NO_HZ_IDLE` which is more appropriate for the use case of HAOS. `CONFIG_NO_HZ_FULL` only makes sense for specific scenarios, as you must also configure CPU affinity and dedicate CPU that will run some (most commonly) real-time task. That is not case of a desktop/server system, where a single CPU processes many tasks. In that case you will only benefit from omitting scheduling-clock interrupts on idle CPUs, which is already achieved through `CONFIG_NO_HZ_IDLE` that also doesn't need any further configuration.

**Comment by unclehack** [comment]
That makes sense. There's no point in trying to do this if such tuning isn't supported. I'll start a new conversation for what may be achievable.
